pages/dashboard/leaderboard.tsx

```diff
@@ -2,11 +2,40 @@ import DashboardLayout from "../../components/layout/DashboardLayout";
 import { parseCookies } from "nookies";
 import { Common } from "../../constants/common";
 import { GetServerSidePropsContext } from "next";
+import { fetchWrapper } from "../../util/httpWrapper";
+import styles from "../../styles/pages/DashboardHome.module.scss";
+import { LeaderboardItem } from "../../components/dashboard/LeaderboardItem";
+import { useEffect, useState } from "react";
 
-function Leaderboard() {
+function Leaderboard({ data }: any) {
+    const [loading, setLoading] = useState(true);
+
+    useEffect(() => {
+        if (data) {
+            setLoading(false);
+        }
+    }, [data]);
+
+    
     return (
         <DashboardLayout title="Leaderboard | ACM at PEC" heading="Leaderboard">
-            <p style={{ textAlign: "center" }}>Coming Soon! 🏆</p>
+            <div className={styles.details}>
+                <div style={{ width: "100%", margin: "2rem" }}>
+                    <div className={styles.contributors}>
+                        <h2 className={styles.header}>🏆 Contributors</h2>
+                        <div className={styles.leaderboard}>
+                            {data && !loading ? (
+                                data
+                                    .filter((user: any) => user.xp > 0)
+                                    .sort((user1: any, user2: any) => user2.xp - user1.xp)
```

Comment:
Data from backend is already sorted so this isnt required ig, what say @harshjohar

Reply:
Yeah makes sense